diff --git a/valid/hudson/build.rb b/valid/hudson/build.rb
index 5388ae0..6fc8202 100755
--- a/valid/hudson/build.rb
+++ b/valid/hudson/build.rb
@@ -128,6 +128,15 @@ $linux_configs = {
       :skip_build => false,
       :valid_platform => [:hw]
     },
+   "k1bio_developer_smp_mmu_rt" => {
+      :arch => "k1b",
+      :core => "k1bio",
+      :board => "developer",
+      :kconfig => "k1bio_developer_smp_mmu_rt_defconfig",
+      :skip_build => false,
+      :valid_platform => [:hw]
+    },
+
   "k1bio_explorer_mmu" => {
     :arch => "k1b",
     :core => "k1bio",
